java/com/engflow/notificationqueue/Client.java

```diff
@@ -0,0 +1,207 @@
+            new StreamObserver<PullNotificationResponse>() {
+              @Override
+              public void onNext(PullNotificationResponse response) {
+                System.out.println("Nothing here");
+                Notification streamedNotification = response.getNotification().getNotification();
+                System.out.println("Notification: " + streamedNotification);
+                Any notificationContent = streamedNotification.getPayload();
+                try {
+                  BuildLifecycleEventNotification lifeCycleEvent =
+                      notificationContent.unpack(BuildLifecycleEventNotification.class);
+                  // Check if this is an invocation started event
+                  if (lifeCycleEvent.getKindCase().name().equals("INVOCATION_STARTED")) {
+                    String invocation = lifeCycleEvent.getInvocationStarted().getInvocationId();
+                    try {
+                      getInvocations(channel, invocation);
+                    } catch (InterruptedException e) {
+                      System.err.println("Could not get invocation with uuid " + invocation);
+                    }
+                  }
+
+                } catch (InvalidProtocolBufferException e) {
+                  throw new RuntimeException(e);
+                }
+              }
+
+              @Override
+              public void onError(Throwable t) {
+                System.err.println("Error on request: " + t.getMessage());
+                finishLatch.countDown();
+              }
+
+              @Override
+              public void onCompleted() {
+                System.out.println("Finished pulling notifications");
+                finishLatch.countDown();
+              }
+            });
+
+    try {
+      requestObserver.onNext(
+          PullNotificationRequest.newBuilder()
+              .setQueue(QueueId.newBuilder().setName(queueName).build())
+              .build());
+    } catch (RuntimeException e) {
+      // Cancel RPC
+      requestObserver.onError(e);
+      throw e;
+    }
+
+    finishLatch.await();
+  }
+
+  public static void getInvocations(ManagedChannel channel, String invocationId)
+      throws InterruptedException {
+    EventStoreGrpc.EventStoreStub asyncStub = EventStoreGrpc.newStub(channel);
+    asyncStub.getInvocation(
+        GetInvocationRequest.newBuilder().setInvocationId(invocationId).build(),
+        new StreamObserver<StreamedBuildEvent>() {
+          @Override
+          public void onNext(StreamedBuildEvent response) {
+            System.out.println("Invocation: " + response.toString());
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            System.err.println("Error on request: " + t.getMessage());
+          }
+
+          @Override
+          public void onCompleted() {
+            System.out.println("Finished pulling invocation");
+          }
+        });
+  }
+
+  private static ManagedChannel createChannel(
```

Comment:
If you go with the `parseOptions` suggestion above, then `createChannel` can just accept an `Options` value, and it won't need to validate arguments, since that will have been done elsewhere.